src/TodoApp.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  FormEvent,
+  useEffect,
+  useState,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { FilterTypes } from './types/FilterTypes';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(() => {
+    const localTodos = localStorage.getItem('todos');
+
+    try {
+      return localTodos ? JSON.parse(localTodos) : [];
+    } catch {
+      return [];
+    }
+  });
+
+  const [title, setTitle] = useState('');
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const filteredTodos = todos.filter(({ completed }) => {
+    switch (pathname) {
+      case FilterTypes.COMPLETED:
+        return completed;
+
+      case FilterTypes.ACTIVE:
+        return !completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const addNewTodo = (event: FormEvent) => {
+    event.preventDefault();
+    const newTitle = title.trim();
+
+    if (!newTitle) {
+      return;
```

Comment:
add hint error for user that title can't be empty 
![image](https://user-images.githubusercontent.com/70009205/206553640-4e2635ff-c617-4b05-a176-cb688b100dc5.png)
now when pressing enter nothing happens and for the user it looks as if nothing is working